src/account.js

```diff
@@ -195,6 +206,16 @@ export class MuxedAccount {
     return this.account.incrementSequenceNumber();
   }
 
+  /**
+   * Creates another muxed "sub"account from the base with a new ID set
+   *
+   * @param  {string} id - the ID of the new muxed account
+   * @return {MuxedAccount} a new instance w/ the specified parameters
+   */
+  createSubaccount(id) {
```

Comment:
If we need this method can we rename this to `createdMuxedAccount`? See https://github.com/stellar/js-stellar-base/issues/433 for why.

However, I think this function confusing. You can't create a muxed account that multiplexes another muxed account. The following code, although verbose, seems much clearer. Is there a reason we can't use it? or a reason we need to be able to create muxed accounts from muxed accounts?

```js
muxedAccount.baseAccount().createMuxedAccount(id)
```

Reply:
Though I completely agree that it's odd to have this method on this object, I added it to `MuxedAccount` so that there was "interface" compatibility with `Account`. And :+1: for the rename, though unfortunate that it wasn't noted at its introduction.

Reply:
What is driving the need for this interface compatibility? I don't see anything new calling `createSubaccount` in this PR.

Reply:
There's no true need (especially since nobody is using muxed accounts); I just had it in mind after js-stellar-sdk#653.

Reply:
Ah, interesting. In this case, I don't think all accounts should have this capability.

Reply:
I agree in theory, but then you cannot use `MuxedAccount` abstraction where `Account` is allowed, yet the two should be largely interchangeable. Fixing this imo fundamentally requires a bigger change (see [this comment](https://github.com/stellar/js-stellar-sdk/pull/655#discussion_r643486317), where `createSubaccount` exists only on the non-interface), but that's out of scope here, so for now I think making muxed accounts "quack like an Account" is a reasonable measure.

Reply:
Got it. Thanks for explaining.

For my own understanding, which is limited here, is there no way in JavaScript/TypeScript to specify an interface type that provides a set of functions, so AccountResponse and MuxedAccount don't always have to have the exact same function set as Account, but rather there could be some interface Account that is a common subset that doesn't include `createSubaccount`?

Reply:
You're essentially describing exactly what I want: an interface that covers the common subset of functionality across `AccountResponse`, `MuxedAccount`, and `Account`. This would involve replacing everywhere that says `Account` (an object) which _used_ to act as said "common subset" with some new interface.